docs/devguide/hexformat.rst

```diff
@@ -67,3 +105,48 @@ Running the ``make all`` command executes the following steps:
 - The ``tools/adduicr.py`` script adds the UICR to the bare hex
 - The final hex file is placed in ``build/firmware.hex``
 - The user can optionally append a script using ``tools/makecombinedhex.py`` (or other tools)
+
+micro:bit V2
+============
+
+This applies to MicroPython for the micro:bit V2, the source of which can be found here: `microbit-foundation/micropython-microbit-v2 <https://github.com/microbit-foundation/micropython-microbit-v2>`_.
+
+This is a port of MicroPython to the micro:bit which uses the CODAL as the underlying target platform.
+
+After cloning this repository update the submodules:
+
+`$ git submodule update --init`
+
+Then build the MicroPython cross-compiler:
+
+`$ make -C lib/micropython/mpy-cross`
+
+After setting up, go to the src/ directory and build:
+
+`$ cd src`
+
+`$ make`
+
+That will build both libmicropython.a (from source in src/codal_port/) and the CODAL app (from source in src/codal_app/). The resulting firmware will be MICROBIT.hex in the src/ directory which can be copied to the micro:bit.
+
+Including a user script
+-----------------------
+
+User scripts are stored in the MicroPython filesystem and if a `main.py` script exists it is run when MicroPython starts. Additional Python scripts can also be included and executed from the main.py file, or the REPL.
+
+The `Python Editor <https://python.microbit.org>`_ uses `microbit-fs <https://github.com/microbit-foundation/microbit-fs>`_ to create the filesystem and include it in the HEX file. The Python Editor must add the filesystem to HEX files for MicroPython V1 & V2, and then combine both into a `Universal HEX <https://tech.microbit.org/software/hex-format/#universal-hex-files>`_ file to ensure compatibility with both hardware variants.
+
+Appended script format (Deprecated)
+-----------------------------------
+
+*This method of appending the script to the end of MicroPython is no longer used. Python files are now stored in the :doc:`filesystem <../filesystem>` and main.py is the program entry point.*
```

Comment:
The URL did not render correctly: 
![image](https://user-images.githubusercontent.com/29712657/106493076-957dbf80-64b0-11eb-8df0-d376a4946616.png)